### ISSUE TITLE
multiclass support

### ISSUE THREAD

**Issue by AlbertDeFusco** [AlbertDeFusco/ScratchNet#1]
let's get the the softmax activation implemented and check that the multiclass cross-entropy loss function and its derivative works

**Comment by timsetsfire** [comment]
I started to do this but there is some reshaping happening in the main network class which stopped me.  We could leave the main network class alone and get this done via vectorization.  What do you think?
